### PR TITLE
Add k-space basics tutorial for MRI reconstruction

### PR DESCRIPTION
**Pull request by ViSaReVe**
git push -u origin add-kspace-basics-tutorial
Adds an introductory notebook covering k-space fundamentals, Fourier transform connection, aliasing from undersampling, and MONAI's reconstruction transforms using the fastMRI knee single-coil dataset. Also adds the missing Reconstruction section to the main README.

Fixes # .

### Description
A few sentences describing the changes proposed in this pull request.

### Checks

- [ ] Avoid including large-size files in the PR.
- [ ] Clean up long text outputs from code cells in the notebook.
- [ ] For security purposes, please check the contents and remove any sensitive info such as user names and private key.
- [ ] Ensure (1) hyperlinks and markdown anchors are working (2) use relative paths for tutorial repo files (3) put figure and graphs in the `./figure` folder
- [ ] Notebook runs automatically `./runner.sh -t <path to .ipynb file>`

## Summary by CodeRabbit

* **New Features**
  * Added an MRI Reconstruction tutorial series, including a K‑space basics notebook with fastMRI knee data and demos for U‑Net and VarNet reconstructions.

* **Documentation**
  * Main README updated: new "Reconstruction" section replaces placeholder.
  * Added tutorials README detailing notebook topics, prerequisites, dataset guidance, and links to related demos.

* **Chores**
  * Test/runner configuration updated to include the new tutorial in execution.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -386,4 +386,12 @@ Example shows the use cases of using MONAI to evaluate the performance of a gene
...
-ECHO�� �����Ǿ� �ֽ��ϴ�.
+
+#### <ins>**Reconstruction**</ins>
+##### [K-Space Basics with fastMRI Knee Data](./reconstruction/MRI_reconstruction/tutorials/01_kspace_basics_fastmri_knee.ipynb)
+This tutorial introduces MRI reconstruction fundamentals: what k-space is, how the Fourier transform connects k-space to images, why undersampling causes aliasing, and how MONAI's reconstruction transforms process k-space data. Uses the fastMRI knee single-coil dataset.
+##### [U-Net MRI Reconstruction](./reconstruction/MRI_reconstruction/unet_demo)
+Training and inference for accelerated MRI reconstruction using BasicUNet on the fastMRI brain multi-coil dataset.
+##### [VarNet MRI Reconstruction](./reconstruction/MRI_reconstruction/varnet_demo)
+Training and inference for accelerated MRI reconstruction using e2e-VarNet on the fastMRI brain multi-coil dataset.
+ECHO�� �����Ǿ� �ֽ��ϴ�.
```
[the remark this answers is not in the document]
Comment (ericspod) [reply README.md]: Please do remove the line starting with "ECHO", I don't know why this was in the original file.

File: reconstruction/MRI_reconstruction/tutorials/01_kspace_basics_fastmri_knee.ipynb [hunk 2]
```diff
@@ -0,0 +1,779 @@
+{
```
Comment (ericspod) [inline reconstruction/MRI_reconstruction/tutorials/01_kspace_basics_fastmri_knee.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #1.    directory = os.environ.get("MONAI_DATA_DIRECTORY")
</pre><p>As was commented by coderabbit, this cell isn't needed since the directory the user will save the validation set to will already exist. </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Project-MONAI/tutorials/pull/2056/discussion/#comment-2804529094'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='6'/>

Comment (ericspod) [inline reconstruction/MRI_reconstruction/tutorials/01_kspace_basics_fastmri_knee.ipynb:1]: <p>Instead this can be <code>data_path = os.path.join(YOUR_DIR_HERE, "knee_singlecoil_val")</code> to indicate  the directory the user has downloaded to has to be put here.</p><p> </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Project-MONAI/tutorials/pull/2056/discussion/#comment-2804529102'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='8'/>

Comment (ericspod) [inline reconstruction/MRI_reconstruction/tutorials/01_kspace_basics_fastmri_knee.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #1.    def as_numpy(x):
</pre><p>We have <a href="https://github.com/Project-MONAI/MONAI/blob/b106a4c35058ebed25e81ba74a51cf4fc79006c8/monai/utils/type_conversion.py#L196" target="_blank">https://github.com/Project-MONAI/MONAI/blob/dev/monai/utils/type_conversion.py#L196</a> for this if you want.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Project-MONAI/tutorials/pull/2056/discussion/#comment-2804529108'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='21'/>

Comment (ericspod) [inline reconstruction/MRI_reconstruction/tutorials/01_kspace_basics_fastmri_knee.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #16.    def ensure_channel_first_2d(x):
</pre><p>We have <code>EnsureChannelFirstd</code> transform for this. </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Project-MONAI/tutorials/pull/2056/discussion/#comment-2804529111'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='24'/>

Comment (ericspod) [inline reconstruction/MRI_reconstruction/tutorials/01_kspace_basics_fastmri_knee.ipynb:1]: <p>As stated this cell isn't needed if the cell creating <code>root_dir</code> is removed. </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Project-MONAI/tutorials/pull/2056/discussion/#comment-2804529116'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='31'/>

**Comment by ericspod** [comment]
Hi @ViSaReVe thanks for the notebook, I think it looks good though I haven't run it myself and it can't as yet be automatically tested. Please look at the comments I made and coderabbit to update a few things. The errors that have come up from the CI jobs is related to [this](https://github.com/pypa/setuptools/issues/5174) I think and we're looking into it.

**Comment by ViSaReVe** [comment]
Thanks for the review @ericspod! I've addressed all comments:

Removed MONAI_DATA_DIRECTORY / root_dir / cleanup cells — now uses data_path = os.path.join("YOUR_DIR_HERE", "knee_singlecoil_val") with error handling
Replaced custom as_numpy() with monai.utils.type_conversion.convert_data_type
Replaced custom ensure_channel_first_2d() with EnsureChannelFirstd(channel_dim="no_channel")
Fixed slice mismatch in Part 4 visualization for multi-slice output

**Review by ericspod — APPROVED** [review]
Hi @ViSaReVe I think it's good now to include. We should be able to get through the CI issues as well. If you wanted you could run the notebook again and not clear the cell output, this would leave plots and code output in place so readers can see what the images are without running the notebook themselves. The file gets larger but that's fine, totally optional though and feel free to merge if you want.

**Comment by ViSaReVe** [comment]
Thanks @ericspod! I've added the cell outputs from a Colab run — the notebook now includes 4 plots (Parts 1, 2, 3, and 6) and key text outputs (data shapes, pipeline shapes) so readers can see the results without running it. Ready to merge when you are!

**Comment by ericspod** [comment]
> Thanks @ericspod! I've added the cell outputs from a Colab run — the notebook now includes 4 plots (Parts 1, 2, 3, and 6) and key text outputs (data shapes, pipeline shapes) so readers can see the results without running it. Ready to merge when you are!

Thanks for the update but a number of the plots look blank (or nearly so). These are mostly the kspace plots but the middle figure of plot 2 I don't think should be so empty. Can anything be done to work on the axis range of these? I think it's just matplotlib not showing things well rather than the data being wrong since the inverse fft works as expected. If you can improve those that would be great, or we can just merge now.

**Comment by ViSaReVe** [comment]
Thanks @ericspod! I've fixed the plot visibility issues:

Plot 2 (center-only) was black — turned out to be a fftshift bug. fastMRI stores k-space with DC already at center, so the extra fftshift was moving DC to the corner and the "center" extraction was grabbing zeros. Fixed by working with k-space directly.
K-space plots were too dark — added percentile-based contrast scaling (vmin/vmax) to all plot cells so the full dynamic range is visible.
Plot 6 ground truth had cross-pattern artifact — the pipeline's reconstruction_esc had wrong shape from FastMRIReader, so now using the ground truth loaded directly from HDF5 instead.
Also removed the garbled ECHO line from README.md.

**Comment by ericspod** [comment]
Hi @ViSaReVe thanks for the continued effort on this notebook. There's a pep8 issue with the cell you've changed to use gdrive, I think you'll only need to move the import line to the first cell.

**Comment by ericspod** [comment]
@coderabbitai propose a committable fix for 01_kspace_basics_fastmri_knee.ipynb to move the google colab import line into the cell with the other imports.

**Comment by ericspod** [comment]
@ViSaReVe it's been a long time but could you come back to this one? We need move only one import in the notebook, which I can't do for fork protection reasons. Thanks!
